src/test/scala/QuartzTypedSchedulerFunctionalSpec.scala

```diff
@@ -0,0 +1,459 @@
+package com.typesafe.akka.extension.quartz
+
+import java.util.logging.Logger
+import java.util.{Calendar, Date}
+
+import akka.actor.testkit.typed.scaladsl.ActorTestKit
+import akka.actor.typed.eventstream.EventStream
+import akka.actor.typed.eventstream.EventStream.Subscribe
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.{ActorRef, Behavior}
+import akka.japi.Option.Some
+import com.typesafe.config.{Config, ConfigFactory}
+import org.junit.runner.RunWith
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+import org.specs2.runner.JUnitRunner
+
+import scala.concurrent._
+import scala.concurrent.duration._
+
+
+@RunWith(classOf[JUnitRunner])
+class QuartzTypedSchedulerFunctionalSpec
+  extends AnyWordSpecLike
+  with Matchers
+  with BeforeAndAfterAll {
+
+  private lazy val testKit = ActorTestKit(SchedulingTypedFunctionalTest.sampleConfiguration)
+  private val _system = testKit.internalSystem
+  private val extension = QuartzSchedulerTypedExtension(_system)
+
+  override protected def afterAll(): Unit = {
+    this.testKit.shutdownTestKit()
+    Await.result(this.testKit.internalSystem.whenTerminated, Duration.Inf)
+  }
+
+  "The Quartz Scheduling Extension" must {
+
+    val tickTolerance = 500 // 500 millisecond tolerance
+
+    "Reject a job which is not named in the config" in {
+      val receiver = testKit.spawn(ScheduleTestReceiver())
+      val probe = testKit.createTestProbe[AnyRef]()
+      receiver ! NewProbe(probe.ref)
+
+      an[IllegalArgumentException] must be thrownBy {
+        `extension`.scheduleTyped("fooBarBazSpamEggsOMGPonies!", receiver, Tick)
+      }
+
+    }
+
+    "Properly Setup & Execute a Cron Job with correct fireTime" in {
+      val receiver = testKit.spawn(ScheduleTestReceiver())
+      val probe = testKit.createTestProbe[AnyRef]()
+      receiver ! NewProbe(probe.ref)
+      val jobDt = `extension`.scheduleTyped("cronEvery10SecondsWithFireTime", receiver, MessageRequireFireTime(Tick))
+
+      /* This is a somewhat questionable test as the timing between components may not match the tick off. */
+      val receipt: Seq[AnyRef] = probe.receiveMessages(5, Duration(50, SECONDS))
+      0 until 5 foreach { i =>
+        val obj = receipt(i) match {
+          case TockWithFireTime(scheduledFireTime, _, _) =>
+            scheduledFireTime
+        }
+        assert(obj === jobDt.getTime + i * 10 * 1000 +- tickTolerance)
+      }
+
+      receipt must have size (5)
+    }
```

Comment:
All of the tests in this class appear to be duplicates of what's in the other test class for the other version of the API. We definitely want to test all of the same cases for both APIs.

My hesitation on approving this PR sooner was that I wanted to look and see if there was a way to avoid duplicating this code. Do you think it's possible to refactor these tests out to either (A) a base testing `trait` or (B) some kind of stateless testing class? Do you see a good way of doing this?

Reply:
I'll think about it how to do it...

Reply:
@enragedginger hey, good afternoon!
I analyzed the point you commented, but note that only the logic is repeated, but the classes are not. They use classes that have no relationship with each other (there is no generation or polymorphism between them). Therefore, I have no way of creating a Trait that can handle classes of different types because unused methods are used.

I recommend that you continue as you are, because you ensure that the same tests as the old version of Akka Actor works with the Akka Typed Actor. In this way, we will show the community that, just as it worked in the past, will work in the future.

What do you think?

Reply:
@enragedginger is there any news about this PR?
Did you consider what I've said?

Reply:
Yeah, that makes sense. Thanks for looking into it!